tools/log_collection/collect_archive_logs.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+# save directory from which command is initiated
+collect_logs_directory=$(pwd)
+
+# make the vmdb/log directory the current directory
+vmdb_logs_directory="/var/www/miq/vmdb"
+pushd ${vmdb_logs_directory}
+
+# eliminiate any prior collected logs to make sure that only one collection is current
+rm -f log/evm_full_archive_$(uname -n)* log/evm_archived_$(uname -n)*
+
+#Source in the file so that we can call postgresql functions
+source /etc/default/evm
+
+tarball="log/evm_archive_$(uname -n)_$(date +%Y%m%d_%H%M%S).tar.xz"
+
+if [[ -n "$APPLIANCE_PG_DATA" && -d "$APPLIANCE_PG_DATA/pg_log" ]]; then
+    echo "This ManageIQ appliance has a Database server and is running version: $(psql --version)"
+    echo " Log collection starting:"
+    XZ_OPT=-9 tar -cJvf ${tarball} --sparse -X $collect_logs_directory/exclude_files BUILD GUID VERSION REGION log/* config/*  /var/log/* log/apache/* $APPLIANCE_PG_DATA/pg_log/* $APPLIANCE_PG_DATA/postgresql.conf
```

Comment:
I agree with pretty with all the changes made.  I do have an additional question.  Do you think this script should be reworked in Ruby since the everything else basically uses it?  The only caveat I found is that I'm not sure we can user `TarWriter` to be as specific as we do with the tar command in the bash script.

Reply:
> Do you think this script should be reworked in Ruby since the everything else basically uses it?

I do, simply because then it could be made into a lib and used in other areas as well (in the UI for example).  I think it is fine to leave it in as a bash script for now, since it is battle tested in it's current form, but think the end goal should be to convert it to something that can be leveraged else where.

> The only caveat I found is that I'm not sure we can user `TarWriter` to be as specific as we do with the tar command in the bash script.

Yeah, the `Gem::TarWriter` only handles the base `tar` file implementation, and doesn't handle compression.  The `rubygems` library leverages `ZLib` from Ruby's standard library to handle `gzip` compression from gems, but that is not what we are using here (I believe it is the `xz` compression algorithm).

We do have a gem that supports `xz` as well, so eventually we should be able to support that if desired.  From what I have noticed, it does seem to provide a smaller file size, but beyond that, I don't know if there are any advantages over one algorithm versus the other.